lib/matplotlib/tests/test_figure.py

```diff
@@ -1456,6 +1456,43 @@ def test_subfigure_pdf():
     buffer = io.BytesIO()
     fig.savefig(buffer, format='pdf')
 
+def test_get_subplot_params():
+    fig = plt.figure()
+    subplotparams_keys = ["left", "bottom", "right", "top", "wspace", "hspace"]
+    subplotparams = fig.get_subplotparams()
+    test_dict = {}
+    for key in subplotparams_keys:
+        attr = getattr(subplotparams, key)
+        assert attr == mpl.rcParams[f"figure.subplot.{key}"]
+        test_dict[key] = attr * 2
+
+    fig.set_subplotparams(test_dict)
+    for key, value in test_dict.items():
+        assert getattr(fig.get_subplotparams(), key) == value
+
+    test_dict['foo'] = 'bar'
+    with pytest.warns(UserWarning,
+                      match="'foo' is not a valid key for set_subplotparams;"
+                      " this key was ignored"):
+        fig.set_subplotparams(test_dict)
+
+    with pytest.raises(TypeError,
+                       match="subplotparams must be a dictionary of "
+                       "keyword-argument pairs or "
+                       "an instance of SubplotParams()"):
+        fig.set_subplotparams(['foo'])
+
+    assert fig.subplotpars == fig.get_subplotparams()
```

Comment:
This test is using `set_subplotparams` which is currently implemented as `set_subplot_params`, and so it is failing.